examples/auto_ml/plot_automl_loop_clean_kata.py

```diff
@@ -0,0 +1,159 @@
+    ]).set_name('LogisticRegression')
+
+    random_forest_classifier = Pipeline([
+        OutputTransformerWrapper(NumpyRavel()),
+        SKLearnWrapper(RandomForestClassifier(), HyperparameterSpace({
+            'n_estimators': RandInt(50, 600),
+            'criterion': Choice(['gini', 'entropy']),
+            'min_samples_leaf': RandInt(2, 5),
+            'min_samples_split': RandInt(1, 3),
+            'bootstrap': Boolean()
+        }))
+    ]).set_name('RandomForestClassifier')
+
+    # Define a classification pipeline that lets the AutoML loop choose one of the classifier.
+    # See also ChooseOneStepOf documentation : https://www.neuraxle.org/stable/api/neuraxle.steps.flow.html#neuraxle.steps.flow.ChooseOneStepOf
+
+    pipeline = Pipeline([
+        FeatureUnion([
+            Pipeline([
+                NumpyFFT(),
+                NumpyAbs(),
+                FeatureUnion([
+                    NumpyFlattenDatum(),  # Reshape from 3D to flat 2D: flattening data except on batch size
+                    FFTPeakBinWithValue()  # Extract 2D features from the 3D FFT bins
+                ], joiner=NumpyConcatenateInnerFeatures())
+            ]),
+            NumpyMean(),
+            NumpyMedian(),
+            NumpyMin(),
+            NumpyMax()
+        ], joiner=NumpyConcatenateInnerFeatures()),
+        # Shape: [batch_size, remade_features]
+        ChooseOneStepOf([
+            decision_tree_classifier,
+            extra_tree_classifier,
+            ridge_classifier,
+            logistic_regression,
+            random_forest_classifier
+        ]),
+        # Shape: [batch_size]
+    ])
+
+    # Create the AutoML loop object.
+    # See also AutoML documentation : https://www.neuraxle.org/stable/api/neuraxle.metaopt.auto_ml.html#neuraxle.metaopt.auto_ml.AutoML
+
+    auto_ml = AutoML(
+        pipeline=pipeline,
+        hyperparams_optimizer=RandomSearchHyperparameterSelectionStrategy(),
+        validation_splitter=ValidationSplitter(test_size=0.20),
+        scoring_callback=ScoringCallback(accuracy_score, higher_score_is_better=False),
+        n_trials=7,
+        epochs=1,
+        hyperparams_repository=InMemoryHyperparamsRepository(cache_folder='cache'),
+        refit_trial=True,
+    )
+
+    # Load data, and launch AutoML loop !
+
+    X_train, y_train, X_test, y_test = load_all_data()
+    auto_ml = auto_ml.fit(X_train, y_train)
+
+    # Get the model from the best trial, and make predictions using predict.
+    # See also predict documentation : https://www.neuraxle.org/stable/api/neuraxle.base.html#neuraxle.base.BaseStep.predict
+
+    best_pipeline = auto_ml.get_best_model()
+    y_pred = best_pipeline.predict(X_test)
+
+    accuracy = accuracy_score(y_true=y_test, y_pred=y_pred)
+    print("Test accuracy score:", accuracy)
+
+
+def load_all_data():
+    # see this repository if you want to execute this example : https://github.com/Neuraxio/Kata-Clean-Machine-Learning-From-Dirty-Code
+    pass
```

Comment:
As an example listed under `/examples`, I would make it work. As a notebook documentation page, it would have been less bad to make the code not compile.